test/geometries/custom_non_copiable/cnc_container.hpp

```diff
@@ -0,0 +1,51 @@
+// Boost.Geometry
+
+// Copyright (c) 2023 Barend Gehrels, Amsterdam, the Netherlands.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+
+#ifndef GEOMETRY_TEST_CNC_CONTAINER_HPP
+#define GEOMETRY_TEST_CNC_CONTAINER_HPP
+
+#include <cstddef>
+#include <deque>
+
+// Define a custom class which is only movable and not copiable, and cannot be indexed.
+// Its derivatives will be linestring, ring, multi*
+// For readability purposes we use the "cnc" abbreviation as "custom_non_copiable"
```

Comment:
This class originates from `all_custom_geometries` but takes it a bit further, making it non-copyable.
For test purposes it has to be indexable, but that's also customized (the library does not see that).

Also the adaptations are more modern (requiring less typedefs, more readable)